src/utils/libfabric/libfabric_rail_manager.cpp

```diff
@@ -179,7 +179,8 @@ nixlLibfabricRailManager::prepareAndSubmitTransfer(
 
         // For TCP providers, use offset 0 instead of virtual address
         // TCP providers don't support FI_MR_VIRT_ADDR and expect offset-based addressing
-        if (data_rails_[rail_id]->provider_name == "tcp" ||
+        if (data_rails_[rail_id]->provider_name == "cxi" ||
+            data_rails_[rail_id]->provider_name == "tcp" ||
```

Comment:
@akkart-aws was there a reason you couldn't just check for FI_MR_VIRT_ADDR directly here?

Reply:
Sure, makes sense. I think it would be simpler and more portable to other providers to just check the bit here instead, if it works. It's a lot quicker to check the one bit instead of these implicit string conversions and comparisons.

Thank you for taking a look.